app/post.go

```diff
@@ -143,6 +147,41 @@ func CreatePost(post *model.Post, channel *model.Channel, triggerWebhooks bool)
 	return rpost, nil
 }
 
+// FillInPostProps should be invoked before saving posts to fill in properties such as
+// channel_mentions.
+//
+// If channel is nil, FillInPostProps will look up the channel corresponding to the post.
+func FillInPostProps(post *model.Post, channel *model.Channel) *model.AppError {
```

Comment:
Maybe we can just skip this whole function if `len(post.ChannelMentions()) == 0`? I'm not sure if that would change much though